File: appambit_sdk_flutter/android/src/main/kotlin/com/appambit/appambit_sdk_flutter/RemoteConfigFlutter.kt
package com.appambit.appambit_sdk_flutter

import android.content.Context
import com.appambit.sdk.RemoteConfig
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodChannel

class RemoteConfigFlutter {
    private lateinit var channel: MethodChannel
    private lateinit var context: Context

    fun attach(binding: FlutterPlugin.FlutterPluginBinding, context: Context) {
        this.context = context
        channel = MethodChannel(binding.binaryMessenger, "com.appambit/remoteconfig")
        channel.setMethodCallHandler { call, result ->
            when (call.method) {
                "enable" -> {
                    val enabled = RemoteConfig.enable()
                    result.success(enabled)
                }
                "getString" -> {
                    val args = call.arguments as? Map<*, *>
                    val key = args?.get("key") as? String
                    if (key == null) {
                        result.error("BAD_ARGS", "Missing 'key'", null)
                    } else {
                        val value = RemoteConfig.getString(key)
                        result.success(value)
                    }
                }
                "getBoolean" -> {
                    val args = call.arguments as? Map<*, *>
                    val key = args?.get("key") as? String
                    if (key == null) {
                        result.error("BAD_ARGS", "Missing 'key'", null)
                    } else {
                        val value = RemoteConfig.getBoolean(key)
                        result.success(value)
                    }
                }
                "getInt" -> {
                    val args = call.arguments as? Map<*, *>
                    val key = args?.get("key") as? String
                    if (key == null) {
                        result.error("BAD_ARGS", "Missing 'key'", null)
                    } else {
                        val value = RemoteConfig.getInt(key)
                        result.success(value)
                    }
                }
                "getDouble" -> {
                    val args = call.arguments as? Map<*, *>
                    val key = args?.get("key") as? String
                    if (key == null) {
                        result.error("BAD_ARGS", "Missing 'key'", null)
                    } else {
                        val value = RemoteConfig.getDouble(key)
                        result.success(value)
                    }
                }
                else -> result.notImplemented()
            }
        }
    }

    fun detach() {
        channel.setMethodCallHandler(null)
    }
}

File: appambit_sdk_flutter/ios/Classes/RemoteConfigFlutter.swift
import Flutter
import UIKit
import AppAmbit

public class RemoteConfigFlutter {
    
    public static func enable(result: @escaping FlutterResult) {
        RemoteConfig.enable()
        result(true)
    }

    public static func disable(result: @escaping FlutterResult) {
        RemoteConfig.disable()
        result(nil)
    }
    
    public static func getString(args: Any?, result: @escaping FlutterResult) {
        guard let argsDict = args as? [String: Any],
              let key = argsDict["key"] as? String else {
            result(FlutterError(code: "BAD_ARGS", message: "Missing or invalid 'key'", details: nil))
            return
        }
        let value = RemoteConfig.getString(key)
        result(value)
    }
    
    public static func getBoolean(args: Any?, result: @escaping FlutterResult) {
        guard let argsDict = args as? [String: Any],
              let key = argsDict["key"] as? String else {
            result(FlutterError(code: "BAD_ARGS", message: "Missing or invalid 'key'", details: nil))
            return
        }
        let value = RemoteConfig.getBoolean(key)
        result(value)
    }
    
    public static func getLong(args: Any?, result: @escaping FlutterResult) {
        guard let argsDict = args as? [String: Any],
              let key = argsDict["key"] as? String else {
            result(FlutterError(code: "BAD_ARGS", message: "Missing or invalid 'key'", details: nil))
            return
        }
        let value = RemoteConfig.getLong(key)
        result(value)
    }
    
    public static func getDouble(args: Any?, result: @escaping FlutterResult) {
        guard let argsDict = args as? [String: Any],
              let key = argsDict["key"] as? String else {
            result(FlutterError(code: "BAD_ARGS", message: "Missing or invalid 'key'", details: nil))
            return
        }
        let value = RemoteConfig.getDouble(key)
        result(value)
    }
}
